scipy/_lib/tests/test__util.py

```diff
@@ -358,9 +367,38 @@ def test_array_api(self, xp, nan_policy):
         elif nan_policy == 'omit' and not is_numpy(xp):
             with pytest.raises(ValueError, match="nan_policy='omit' is incompatible"):
                 _contains_nan(x, nan_policy)
+            assert _contains_nan(x, nan_policy, xp_omit_okay=True)
         elif nan_policy == 'propagate':
             assert _contains_nan(x, nan_policy)
 
+    @pytest.mark.skip_xp_backends("numpy", reason="lazy backends only")
+    @pytest.mark.skip_xp_backends("cupy", reason="lazy backends only")
+    @pytest.mark.skip_xp_backends("array_api_strict", reason="lazy backends only")
+    @pytest.mark.skip_xp_backends("torch", reason="lazy backends only")
+    def test_array_api_lazy(self, xp):
+        rng = np.random.default_rng(932347235892482)
+        x0 = rng.random(size=(2, 3, 4))
+        x = xp.asarray(x0)
+
+        xp_assert_equal(_contains_nan(x), xp.asarray(False))
+        xp_assert_equal(_contains_nan(x, "propagate"), xp.asarray(False))
+        xp_assert_equal(_contains_nan(x, "omit", xp_omit_okay=True), xp.asarray(False))
+        # Lazy arrays don't support "omit" and "raise" policies
+        with pytest.raises(TypeError):
```

Comment:
TODO get an error message that's comprehensible to the final user. Blocked by https://github.com/data-apis/array-api-compat/pull/228

Reply:
No, there is much more widespread work that needs to happen downstream of that PR

Reply:
Added TODO to the code